Expand and re-organize InfluxDB Enterprise authn and authz docs (#3330)
influxdata/docs-v2: content/enterprise_influxdb/v1.9/features/_index.md

    weight: 60
---

InfluxDB Enterprise has additional capabilities that enhance
[availability](#clustering),
[scalability](#clustering), and
[security](#security),

> as well as providing [eventual consistency](#eventual-consistency).

Reviewer: ```suggestion
and provide [eventual consistency](#eventual-consistency).
```

## Clustering

InfluxDB Enterprise runs on a network of independent servers, a *cluster*,
to provide fault tolerance, availability, and horizontal scalability of the database.

> Many InfluxDB Enterprise features are available
> when run with a single meta node and a single data node
> However this does not take advantage of the clustering capablity,

Reviewer: ```suggestion
While many InfluxDB Enterprise features are available
when run with a single meta node and a single data node, this configuration does not take advantage of the clustering capablity
```

> and therefore does not provide high availablity assurances.

Reviewer: ```suggestion
or ensure high availablity.
```

Nodes can be added to an existing cluster to improve database performance for querying and writing data.
Certain configurations (e.g., 3 meta and 2 data node) provide high-availability assurances
while making certain tradeoffs in query peformance when compared to a single node.

> Further increasing the number of nodes can improve performance in both respects

Reviewer: ```suggestion
Further increasing the number of nodes can improve performance in both respects.
```

> For example, a cluster with 4 data nodes and a [replication factor]() of 2

Reviewer: link to replication factor?

can support a higher volume of write traffic than a single node could.
It can also support a higher *query* workload, as the data is replicated
in two locations. Performance of the queries may be on par with a single
node in cases where the query can be answered directly by the node which
receives the query.

For more information on clustering, see [Clustering in InfluxDB Enterprise](/enterprise_influxdb/v1.9/concepts/clustering/).

## Security

Enterprise authorization uses an expanded set of [*16 user permissions and roles*](/enterprise_influxdb/v1.9/features/users/).
(InfluxDB OSS only has `READ` and `WRITE` permissions.)
Administrators can give users permission to read and write to databases,

> create and remove databases, rebalance a cluster, and manage particular resources.

Reviewer: this covers the conceptual overview of permissions--we probably don't need a separate doc (and can keep the Configure/Manage sections strictly procedural content

Organizations can automate managing permissions with the [InfluxDB Enterprise Meta API](/enterprise_influxdb/v1.9/administration/manage/security/authentication_and_authorization-api/).

[Fine-grained authorization](/enterprise_influxdb/v1.9/guides/fine-grained-authorization/)
for particular data is also available.

InfluxDB Enterprise can also use [LDAP for managing authentication](/enterprise_influxdb/v1.9/administration/manage/security/ldap/).

For FIPS compliance, InfluxDB Enterprise password hashing alogrithms are configurable.

{{% note %}}
Kapacitor OSS can also delegate its LDAP and security setup to InfluxDB Enterprise.

> For details, see ["Set up InfluxDB Enterprise authorizations"](https://test2.docs.influxdata.com/kapacitor/latest/administration/auth/influxdb-enterprise-auth/).

Reviewer: ```suggestion
For details, see ["Set up InfluxDB Enterprise authorizations"](/kapacitor/latest/administration/auth/influxdb-enterprise-auth/).
```

{{% /note %}}

## Eventual consistency

### Hinted handoff

Hinted handoff (HH) is how InfluxDB Enterprise deals with data node outages while writes are happening.
HH is essentially a durable disk based queue.

For more information, see ["Hinted handoff"](/enterprise_influxdb/v1.9/concepts/clustering/#hinted-handoff).

### Anti-entropy

> Anti-entropy is an optional services to eliminate edge cases related to cluster consistency.

Reviewer: ```suggestion
Anti-entropy is an optional service to eliminate edge cases related to cluster consistency.
```
